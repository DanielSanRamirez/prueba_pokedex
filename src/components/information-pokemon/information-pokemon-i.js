import React, { useEffect, useState } from 'react'
import { getP } from '../../helper/getP'
import { InformationPokemon } from './information-pokemon'

export const InformationPokemonI = (props) => {

    const {
        idPokemon
    } = props

    const url = `https://pokeapi.co/api/v2/pokemon/${idPokemon}/`

    const [namePokemon, setNamePokemon] = useState()
    const [typesPokemon, setTypesPokemon] = useState()
    const [weightPokemon, setWeightPokemon] = useState()

    useEffect(() => {
        fetchEachPokemon()
    }, [idPokemon])

    const fetchEachPokemon = async () => {
        const dataO = await getP(url)
        setNamePokemon(dataO.name)
        const types = dataO.types.map(t => t.type.name)
        setTypesPokemon([types])
        setWeightPokemon(dataO.weight)
    }

    return (
        <div>
            {
                (namePokemon === undefined && idPokemon === undefined && typesPokemon === undefined && weightPokemon === undefined) ?
                    'Cargando...'
                    :
                    <InformationPokemon namePokemon={namePokemon} idPokemon={idPokemon} typesPokemon={typesPokemon} weightPokemon={weightPokemon} />
            }

        </div>
    )
}
