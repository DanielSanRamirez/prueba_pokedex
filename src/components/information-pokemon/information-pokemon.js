import React from 'react'

export const InformationPokemon = (props) => {

    const {
        namePokemon, idPokemon, typesPokemon, weightPokemon
    } = props

    const urlImage = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/`

    return (
        <div>
            {
                typesPokemon === undefined ? 'Cargando...' :
                    <div>
                        <img src={`${urlImage}${idPokemon}.png`} alt='' />
                        <h3># {idPokemon}</h3>
                        <h3 id="namePokemon">{namePokemon}</h3>
                        <h3>Types</h3>
                        <p>{typesPokemon[0].map(t => t + ' ')} </p>
                        <h3>Peso</h3>
                        <p>{weightPokemon} kg</p>
                    </div>
            }


        </div>
    )
}
