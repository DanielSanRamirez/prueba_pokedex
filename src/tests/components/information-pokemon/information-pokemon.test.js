import { shallow } from "enzyme"
import { InformationPokemon } from "../../../components/information-pokemon/information-pokemon"

describe('Test in component <InformationPokemon />', () => {

    const namePokemon = 'bulbasaur'
    const idPokemon = 1
    const typesPokemon = [['grass', 'poison']]
    const weightPokemon = 69
    const srcImage = "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"

    const wrapper = shallow(<InformationPokemon namePokemon={namePokemon} idPokemon={idPokemon} typesPokemon={typesPokemon} weightPokemon={weightPokemon} />)

    test('should be a napshots', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('should have a title with the image', () => {

        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(srcImage)

    })
    
    test('should have a Pokemons name', () => {
        
        const name = wrapper.find('h3').at(1).text()
        expect(name).toBe(namePokemon)

    })
    

})
