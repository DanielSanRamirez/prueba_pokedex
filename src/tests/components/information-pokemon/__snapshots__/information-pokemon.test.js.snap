// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <InformationPokemon /> should be a napshots 1`] = `
<div>
  <div>
    <img
      alt=""
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    />
    <h3>
      # 
      1
    </h3>
    <h3
      id="namePokemon"
    >
      bulbasaur
    </h3>
    <h3>
      Types
    </h3>
    <p>
      grass 
      poison 
       
    </p>
    <h3>
      Peso
    </h3>
    <p>
      69
       kg
    </p>
  </div>
</div>
`;
